### PR TITLE
NASS-736: Admin panel settings editing

### PR DESCRIPTION
**Pull request by tcodling**
### Changes

This PR adds the basic ability to view and edit the settings table through the admin panel. I have added a JSON editor used to display and edit the settings in JSON form. I have added a new get route that will 'get' groups of settings by scope/facility and also a 'put' route that updates settings by scope/facility

This is only the most basic ui for now without input from Felicity. This seemed like the simplest solution but am open to tweaks if the reviewer has any ideas! 

I also wonder if I have managed my state in the most effective way and am open to any suggestions if this seems hard to follow or could be more concise.

The way it works is there is a new sidebar button in the admin panel called "Settings" that opens a JSON view of the settings for the deployment. Here there is a drop down with two predefined options always there (`Central server` and `All facilities`) and the rest is populated with linked facilities to the deployment. You can switch between these options to view and save the settings relevant to that server/facility
- `Central` - these settings aren't synced to any facility
- `Global` - these settings are synced to all facilities
- `Facility` - these settings are ONLY synced to the specific facility defined in the `facility_id` column

image
image
image

### Checklist

- [ ] Code is finished
- [ ] Tests are written
- [ ] UI Screenshots added to the Linear issue
- [ ] Testing notes added to the Linear issue

_Upon merging:_

- [ ] Update the changelog (find it [here](https://github.com/beyondessential/tamanu/releases) in the appropriate _draft_ release)
  - [ ] with a ticket reference (e.g. `TAN-123: a one line description`, keep the lists sorted)
  - [ ] any config/settings changes and manual deployment steps
  - [ ] any db schema or other changes that could impact downstream data analysis
- [ ] Update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly) or [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q) as needed
- [ ] Update the [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c) as needed

### CODE REVIEW

**Review by edmofro — CHANGES_REQUESTED** [review]
Amazing work on this! Just submitting the comments from our walk through, haven't looked at the back end side yet

**Review by dannash100 — COMMENTED** [review]
Woo fantastic work on this boss, mainly just suggestions and nit picks,
Main thing i think is to remove the mixed types of that `selectedFacility` state.
I rekon the formik consideration can be ignored for now to keep scope in check! 💯

**Review by dannash100 — COMMENTED** [review]
Much better imho! Will tick when update the request to the try catch

**Review by dannash100 — APPROVED** [review]
